app/components/channel_item/channel_item.tsx

```diff
@@ -0,0 +1,265 @@
+        borderColor: theme.centerChannelBg,
+    },
+    mutedBadge: {
+        opacity: 0.4,
+    },
+    activeItem: {
+        backgroundColor: changeOpacity(theme.sidebarTextActiveColor, 0.1),
+        borderLeftColor: theme.sidebarTextActiveBorder,
+        borderLeftWidth: 5,
+        marginLeft: 0,
+        paddingLeft: 14,
+    },
+    textActive: {
+        color: theme.sidebarText,
+    },
+    teamName: {
+        color: changeOpacity(theme.centerChannelColor, 0.64),
+        paddingLeft: 12,
+        marginTop: 4,
+        ...typography('Body', 75),
+    },
+    teamNameTablet: {
+        marginLeft: -12,
+        paddingLeft: 0,
+        marginTop: 0,
+        paddingBottom: 0,
+        top: 5,
+    },
+}));
+
+const textStyle = StyleSheet.create({
+    bright: typography('Body', 200, 'SemiBold'),
+    regular: typography('Body', 200, 'Regular'),
+});
+
+const ChannelListItem = ({
+    channel, collapsed, currentUserId, hasDraft,
+    isActive, isInfo, isMuted, isVisible,
+    myChannel, onPress, teamDisplayName, testID}: Props) => {
+    const {formatMessage} = useIntl();
+    const theme = useTheme();
+    const isTablet = useIsTablet();
+    const styles = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+
+    const teammateId = (channel.type === General.DM_CHANNEL) ? getUserIdFromChannelName(currentUserId, channel.name) : undefined;
+    const isOwnDirectMessage = (channel.type === General.DM_CHANNEL) && currentUserId === teammateId;
+
+    // Make it brighter if it's not muted, and highlighted or has unreads
+    const isUnread = !isMuted && (myChannel && (myChannel.isUnread || myChannel.mentionsCount > 0));
+
+    const shouldCollapse = (collapsed && !isUnread) && !isActive;
+    const sharedValue = useSharedValue(shouldCollapse);
+    const height = useMemo(() => {
+        let h = 40;
+        if (isInfo) {
+            h = (teamDisplayName && !isTablet) ? 58 : 44;
+        }
+        return h;
+    }, [teamDisplayName, isInfo, isTablet]);
+
+    useEffect(() => {
+        sharedValue.value = shouldCollapse;
+    }, [shouldCollapse]);
+
+    const animatedStyle = useAnimatedStyle(() => {
+        return {
+            marginVertical: withTiming(sharedValue.value ? 0 : 2, {duration: 500}),
+            height: withTiming(sharedValue.value ? 0 : height, {duration: 500}),
+            opacity: withTiming(sharedValue.value ? 0 : 1, {duration: 500, easing: Easing.inOut(Easing.exp)}),
+        };
+    }, [teamDisplayName, isTablet, isInfo, height]);
+
+    const switchChannels = useCallback(() => {
+        if (myChannel) {
+            if (onPress) {
+                onPress(myChannel.id);
+            } else {
+                switchToChannelById(serverUrl, myChannel.id);
+            }
+        }
+    }, [myChannel?.id, serverUrl]);
+
+    const membersCount = useMemo(() => {
+        if (channel.type === General.GM_CHANNEL) {
+            return channel.displayName.split(',').length;
```

Comment:
User First Name (and probably others) may have `,`, so if the teammateDisplay is set to full name, it will show the , in the display name, which will break this count.

0/5 whether we should plan for that corner case.

Reply:
changed to channel.name